[#58337] use correct link attachment upload

Kharonus opened the pull request:
# Ticket
[OP#58337](https://community.openproject.org/work_packages/58337)

# What are you trying to accomplish?
- if attachments are disabled in files tab, file upload inline in rich text editors should stil work as expected

# What approach did you choose and why?
- use `addAttachment` link as an upload target, if no direct upload is specified - before was `attachments` link, which is conditionally hidden

Kharonus committed:
[#58337] use correct link attachment upload

- https://community.openproject.org/work_packages/58337
- use `addAttachment` link as an upload target, if no direct upload is
  specified - before was `attachments` link, which is conditionally
  hidden

Kharonus committed:
[#58337] fixed upload target for new resources

oliverguenther commented:
For this kind of bug, it sounds like a good idea to add a spec to it. Could you please add one to make sure it's regression-safe? 🙏

Kharonus committed:
[#58337] fix issue with custom text editor

- now the ckeditor instance of a custom text widget in a grid has access
  to the correct attachments link

Kharonus committed:
[#58337] add feature spec for image upload in activities

Kharonus committed:
[#58337] disabled cuprite for image upload spec

akabiru commented on spec/features/activities/work_package/activities_spec.rb:
```diff
@@ -901,6 +901,32 @@
+        page.find_test_selector("op-submit-work-package-journal-form").click
+
+        expect(find_test_selector("op-journal-notes-body")).to have_css("img")
+        expect(page).to have_test_selector("op-journal-detail-description", text: "File image.png added as attachment")
```
✨

akabiru reviewed (approved):
🥇
